atc/config.go

```diff
@@ -40,6 +43,24 @@ func (groups GroupConfigs) Lookup(name string) (GroupConfig, int, bool) {
 	return GroupConfig{}, -1, false
 }
 
+type VarSourceConfig struct {
+	Name   string      `json:"name"`
```

Comment:
Actually, is this name used anywhere?

It seems like it would make sense to be able to explicitly reference the var source to fetch from, e.g. `((myvault:foo/bar.baz))`. But I don't think that's been implemented.

I think we should either implement that or remove the field for now.

Reply:
`((myvault:foo/bar.baz))` is exactly what one of our users asked to me when I demoed this feature to him. I plan to implement that with a separate PR as I want to make this feature work first, and don't want this PR to grow any further. :)

Reply:
@evanchaoli is there a strong use case for wanting to cycle through all cred managers? It seems like we could get into some weird behaviour if someone added a new cred manager which unknowingly overrides a value from another one. And they wouldn't know it until it was already running in the pipeline.

I know it adds more to this PR, but I would be in favour of referencing cred managers by name as being the only supported method. It's much more in line with how resources work, and the direction we want to take with re-usable "resources/tasks" in `projects`.

Reply:
@pivotal-jwinters I don't have a strong use case right now. One use case I can image is, it would be helpful for pipeline debugging. Like when debugging the pipeline, user may add a `dummy` var source in the first place, which would make sure all vars can be interpolated.

The other point is that, I don't think much user would add more than one var_source to a pipeline. Practically, maintaining a Vault is not easy given the serious security matter. So I don't think cycle through all var sources would bring serious issue. Only when users start to use the feature, then we know if that's really an issue.

To support `((myvault:foo/bar.baz))` syntax, some more changes are needed, because with current code, var_source name is not tracked by `creds.Secrets`. Again, I don't want to make this PR any bigger, it has been functional, and should fit most of use cases.

As our communication has long delay due to time difference, I just thought over, if you strongly insist to support var source name, a quick trade-off I can do is, to restrict only the first var_source defined in pipeline will be used. This trade-off would 1) eliminate the "cycle through" concern, 2) not impact our use case because our use will mostly add only one var_source to a pipeline, 3) still support the debugging use case as I described above.

To summarize, I still suggest merge this PR first. If you are strongly against that, I can do the trade-off. Once this PR is merged, I commit to contribute a PR to support `((myvault:foo/bar.baz))` syntax.

Reply:
@evanchaoli I understand that it's functional and that it meets all your use cases, however if we introduce named credential managers later on, it will be a breaking change. However since since this is going to be an experimental/undocumented feature maybe that's ok. I think as long as we get to the point where named credential managers are the only supported method before we remove the experimental tag, then I would be ok merging this as is.

Reply:
👍 to merging for now and requiring named var references later - I'm working that aspect into my RFC. 🙂

Reply:
@pivotal-jwinters Thanks for understanding. I'm absolutely fine with only removing the experimental tag after named syntax is supported. :) But to clarify, the next PR, for named var source, would NOT be a breaking change. The behavior I'm planning to implement is like:

* `((var1))`: iterate over all var_sources including the system level credential manager to look for `var1`
* `((name1:var1))`:  only look up var source `name1` for `var1`.
* `((name1,[name2,...]:var1))`: only look up the multiple specified var sources for `var1`.
* `((sys:var1))`: only look up the system level credential manager for `var1`
* `((name1,[name2,...,]sys:var1))`: only look up specified var sources as well as the system level credential manager for `var1`.

which would leave the flexibility to users. 

Anyway, please merge this PR first. If you have different opinion with the design, we may discuss in a new issue. Once this PR is merged, I'll create an issue to track the named var source change, and we may discuss the design over there.